test/Microsoft.AspNet.Mvc.Core.Test/ModelBinding/CollectionModelBinderTest.cs

```diff
@@ -459,7 +451,7 @@ private class ListWithThrowingConstructor<T> : List<T>
             public ListWithThrowingConstructor()
                 : base()
             {
-                throw new ApplicationException("No, don't do this.");
+                throw new RankException("No, don't do this.");
```

Comment:
Quite likely that a Task.Result can throw `AggregateException`. This ensures it's unique.